pyhik/hikvision.py

```diff
@@ -285,16 +288,32 @@ def get_device_info(self):
 
         try:
             response = self.hik_request.get(url)
+            if response.status_code == requests.codes.unauthorized:
+                _LOGGING.debug('Basic authentication failed. Using digest.')
+                self.hik_request.auth = HTTPDigestAuth(self.usr, self.pwd)
+                response = self.hik_request.get(url)
```

Comment:
I don't think you need this twice.  I'd do the 404 check first, that should get you the correct url, then check for the unauthorized code and switch the auth as needed.

Reply:
I think authorization/authentication happens before it validates the url?

In postman, if I pop in an invalid url and an invalid password, I get a 401 telling me I don't have permission for the invalid url.

I could probably change it so that if I get a 401, switch to digest and try again. If I get a 404, then try the 2nd url assuming that to get a 404, I had to successfully authenticate first.

Reply:
I don't see those results when I do checks with Postman.  I figure the URL needs to exist in order to generate any kind of authentication response.  Make sure in your tests you are using a completely invalid URL.  For example, my camera will return a response (usually 403) for any url with an IASPI prefix, but as soon as I alter then endpoint to IASPI2 or similar I get a 404 regardless of the auth details.